streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2348,266 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
         assertThat(exception.getMessage(), equalTo("named can't be null"));
     }
 
+
+    @Test
+    public void shouldKTableFromKStream() {
```

Comment:
Can we add one more check to this test, to verify that no state store is created for the `KTable`? Ie, we should get the `TopologyDescription` and verify that there is no state store attached. Example: https://github.com/apache/kafka/blob/trunk/streams/src/test/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImplTest.java#L208-L243

This this case, I would recommend to rename this this to `shouldNotMaterializedKTableFromKStream`